tools/apiview/parsers/rust-api-parser/clean-rust-doc-output/src/main.rs

```diff
@@ -0,0 +1,37 @@
+use serde_json::Value;
+use std::fs::File;
+use std::io::{self, Read, Write};
+
+fn main() -> io::Result<()> {
+    // Read the JSON file generated by rustdoc
+    let mut file = File::open("inputs/docs.json")?;
+    let mut data = String::new();
+    file.read_to_string(&mut data)?;
+
+    // Parse the JSON data
+    let v: Value = serde_json::from_str(&data).expect("Unable to parse JSON");
+
+    // Extract the exported API surface
+    let mut exported_api = serde_json::Map::new();
+    exported_api.insert("crate_version".to_string(), v["crate_version"].clone());
+    exported_api.insert("includes_private".to_string(), v["includes_private"].clone());
```

Comment:
All this works, but since you're importing the `derive` feature, I expected to see models. In the end, it'll probably make things easier - not just to read, but maintain. And your models only need to declare what you need; everything else will just be dropped. ultimately, it's going to be a lot less code. For all this parsing you're doing, for example, would just be a couple models and you only deserialize it all in one line.

Reply:
this is a great suggestion, I'll follow this to get a concise api-json.